gammapy/image/maps.py

```diff
@@ -499,6 +500,7 @@ def write(self, filename=None, header=None, **kwargs):
         hdulist = fits.HDUList()
         for name in self.get('_map_names', sorted(self)):
             hdu = self[name].to_image_hdu()
+            hdu.name = name
```

Comment:
Actually this shouldn't be necessary, because the name is already passed to the `ImageHDU` object in `to_image_hdu()`. Can you explain why this is needed?